lib/experimental/Lists/DataList/ItemWithCopyButton.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react"
+
+import { Icon } from "@/components/Utilities/Icon"
+import { ItemProps } from "@/experimental/Lists/DataList"
+import { ItemContainer } from "@/experimental/Lists/DataList/ItemContainer"
+import { CheckCircle, LayersFront } from "@/icons"
+import { cn } from "@/lib/utils"
+
+const COPIED_SHOWN_MS = 750
+
+type CopyButtonProps = ItemProps
+
+export const ItemWithCopyButton = ({
+  text,
+  icon: IconComponent,
+}: CopyButtonProps) => {
+  const [copied, setCopied] = useState(false)
+
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => setCopied(false), COPIED_SHOWN_MS)
+
+      return () => clearTimeout(timer)
+    }
+  }, [copied])
+
+  const copyHandler = async () => {
+    try {
+      await navigator.clipboard.writeText(text)
+      setCopied(true)
+    } catch (error) {
+      void error
+    }
+  }
+
+  return (
+    <ItemContainer
+      as="button"
+      text={text}
+      leftIcon={IconComponent}
+      onClick={copyHandler}
+      className={
+        copied
+          ? "transition-colors duration-300 hover:bg-f1-background-positive focus-visible:bg-f1-background-positive"
+          : undefined
+      }
+      aria-label={copied ? "Copied!" : `Copy ${text}`}
```

Comment:
What's the benefit of doing this? The `Copied!` text won't reach the user. Also, it is hardcoded in english 🤔